packages/eslint-plugin/src/rules/typedef.ts

```diff
@@ -133,87 +147,86 @@ export default util.createRule<[Options], MessageIds>({
 
     function isVariableDeclarationIgnoreFunction(node: TSESTree.Node): boolean {
       return (
-        !!options[OptionKeys.VariableDeclarationIgnoreFunction] &&
```

Comment:
I moved this part to line 175 (so we can avoid enter `ClassProperty`) + line 228.